conformance/lang/expressions/multiplicative-expr/division.balt

```diff
@@ -4033,3 +4033,922 @@ Description: Test integer division panic at module-level.
+}
+
+Test-Case: output
+Description: Test float division at module-level.
+Labels: DecimalNumber, float, float:Infinity, float:NaN, int, multiplicative-expr
+
+int a = 2;
+float b = float:Infinity;
+float c = 4.5e-1;
+float d = float:NaN;
+
+float e = b / a;
+float f = c / a;
+float g = d / a;
+
+function init() {
+    io:println(e); // @output Infinity
+    io:println(f); // @output 0.225
+    io:println(g); // @output NaN
+}
+
+Test-Case: output
+Description: Test float division with constants.
+Labels: DecimalNumber, const-expr, float, float:Infinity, float:NaN, int, module-const-decl, multiplicative-expr
+
+const int constInt = 2;
+const float constFloat = 0.45;
+
+const float r = constFloat / constInt;
+
+function init() {
+    int a = 2;
+    float b = float:Infinity;
+    float c = 4.5e-1;
+    float d = 0;
+    float e = float:NaN;
+    int f = int:MAX_VALUE;
+    int g = int:MIN_VALUE;
+
+    io:println(r); // @output 0.255
+
+    float v = b / constInt;
+    io:println(v); // @output Infinity
+
+    v = c / constInt;
+    io:println(v); // @output 0.255
+
+    v = d / constInt;
+    io:println(v); // @output 0.0
+
+    v = e / constInt;
+    io:println(v); // @output NaN
+
+    v = constFloat / a;
+    io:println(v); // @output 0.255
+
+    v = constFloat / f;
+    io:println(v); // @output 4.87890977618477E-20
+
+    v = constFloat / g;
+    io:println(v); // @output -4.87890977618477E-20
+
+    v = constFloat / constInt;
+    io:println(v); // @output 0.225
+}
+
+Test-Case: output
+Description: Test float division with operands of different int subtypes.
+Labels: DecimalNumber, float, int, multiplicative-expr
+
+type Ints -9|9;
+
+function init() {
+    int:Signed8 a = -2;
+    int:Signed16 b = 2;
+    int:Signed32 c = -4;
+    int:Unsigned8 d = 4;
+    int:Unsigned16 e = 5;
+    int:Unsigned32 f = 10;
+    byte g = 25;
+
+    float h = 2.5;
+    Ints i = 9;
+
+    float v = h / a;
+    io:println(v); // @output -1.25
+
+    v = h / b;
+    io:println(v); // @output 1.25
+
+    v = h / c;
+    io:println(v); // @output -0.625
+
+    v = h / d;
+    io:println(v); // @output 0.625
+
+    v = h / e;
+    io:println(v); // @output 0.5
+
+    v = h / f;
+    io:println(v); // @output 0.25
+
+    v = h / g;
+    io:println(v); // @output 0.1
+
+    v = h / i;
+    io:println(v); // @output 0.2777777777777778
+}
+
+Test-Case: output
+Description: Test float division with optional integers.
+Labels: DecimalNumber, float, float-NaN, int, multiplicative-expr, nil-literal, numeric-nil-lifting
+
+function init() {
+    int? a = 4;
+    int? b = ();
+    float c = 4.5e-1;
+    float d = float:NaN;
+
+    float? v = c / a;
+    io:println(v); // @output 0.1125
+
+    v = c / b;
+    io:println(v); // @output ()
+
+    v = d / a;
+    io:println(v); // @output NaN
+
+    v = d / b;
+    io:println(v); // @output ()
+}
+
+Test-Case: output
+Description: Test optional float division with optional and non-optional integers.
```

Comment:
For this, we should cover. 

- optional float and non optional int
- optional float and optional int
- non optional float and non optional int
- non optional float and optional int

Some you have covered already. May be we can split them into four separate tests